framework/src/controller/controller.js

```diff
@@ -152,9 +150,18 @@ class Controller {
 		// eslint-disable-next-line no-restricted-syntax
 		for (const alias of Object.keys(modules)) {
 			const { klass, options } = modules[alias];
-			if (options.useSocketChannel) {
-				// eslint-disable-next-line no-await-in-loop
-				await this._loadModuleWithSocketChannel(alias, klass, options);
+
+			if (options.loadAsChildProcess) {
+				if (this.config.ipc.enabled) {
+					// eslint-disable-next-line no-await-in-loop
+					await this._loadChildProcessModule(alias, klass, options);
+				} else {
+					this.logger.warn(
+						`IPC is disabled. ${alias} will be loaded in-memory.`
+					);
+					// eslint-disable-next-line no-await-in-loop
+					await this._loadInMemoryModule(alias, klass, options);
+				}
```

Comment:
maybe if else statements can be simplified with using `continue;` ?

Reply:
We have a eslint rule `no-continue`. That's why I didn't use it.
We shouldn't be using `for of` with `await` anyway.
Next iteration we can improve this.

Reply:
> We shouldn't be using for of with await anyway.
Not related to this but what's the reason for this? i think this is the only way though

Reply:
afaik `for await...of` was created for this